--- easyrsa.orig	2019-05-13 21:16:32.648330533 +0200
+++ easyrsa	2019-05-13 21:19:49.430776861 +0200
@@ -215,6 +215,7 @@
 Certificate & Request options: (these impact cert/req field values)
 
 --days=#        : sets the signing validity to the specified number of days
+--enddate=#     : sets the signing validity (Not After) to the specified date
 --digest=ALG    : digest to use in the requests & certificates
 --dn-mode=MODE  : DN mode to use (cn_only or org)
 --keysize=#     : size in bits of keypair to generate
@@ -807,9 +808,16 @@
 	# sign request
 	# shellcheck disable=SC2086
 	crt_out_tmp="$(mktemp "$crt_out.XXXXXXXXXX")"; EASYRSA_TEMP_FILE_2="$crt_out_tmp"
-	"$EASYRSA_OPENSSL" ca -utf8 -in "$req_in" -out "$crt_out_tmp" -config "$EASYRSA_SAFE_CONF" \
-		-extfile "$EASYRSA_TEMP_EXT" -days "$EASYRSA_CERT_EXPIRE" -batch $opts \
-		|| die "signing failed (openssl output above may have more detail)"
+	if [ -n "${EASYRSA_CERT_END_DATE}" ]
+	then
+		"$EASYRSA_OPENSSL" ca -utf8 -in "$req_in" -out "$crt_out_tmp" -config "$EASYRSA_SAFE_CONF" \
+			-extfile "$EASYRSA_TEMP_EXT" -enddate "$EASYRSA_CERT_END_DATE" -batch $opts \
+			|| die "signing failed (openssl output above may have more detail)"
+	else
+		"$EASYRSA_OPENSSL" ca -utf8 -in "$req_in" -out "$crt_out_tmp" -config "$EASYRSA_SAFE_CONF" \
+			-extfile "$EASYRSA_TEMP_EXT" -days "$EASYRSA_CERT_EXPIRE" -batch $opts \
+			|| die "signing failed (openssl output above may have more detail)"
+	fi
 	mv "$crt_out_tmp" "$crt_out"; EASYRSA_TEMP_FILE_2=
 	notice "\
 Certificate created at: $crt_out
@@ -1565,6 +1573,9 @@
 		export EASYRSA_CA_EXPIRE="$val"
 		export EASYRSA_CRL_DAYS="$val"
 		;;
+	--enddate)
+		export EASYRSA_CERT_END_DATE="$val"
+		;;
 	--pki-dir)
 		export EASYRSA_PKI="$val" ;;
 	--use-algo)
