diff --git a/easy-rsa/easyrsa b/easy-rsa/easyrsa
index e004e2b..01b2f64 100755
--- a/easy-rsa/easyrsa
+++ b/easy-rsa/easyrsa
@@ -190,6 +190,7 @@ General options:
 Certificate & Request options: (these impact cert/req field values)
 
 --days=#        : sets the signing validity to the specified number of days
+--enddate=#     : sets the signing validity (Not After) to the specified date
 --digest=ALG    : digest to use in the requests & certificates
 --dn-mode=MODE  : DN mode to use (cn_only or org)
 --keysize=#     : size in bits of keypair to generate
@@ -658,9 +659,17 @@ $EASYRSA_TEMP_FILE"
 
 	# sign request
 	crt_out_tmp="$(mktemp -u "$crt_out.XXXXXXXXXX")"; EASYRSA_TEMP_FILE_2="$crt_out_tmp"
-	"$EASYRSA_OPENSSL" ca -utf8 -in "$req_in" -out "$crt_out_tmp" -config "$EASYRSA_SSL_CONF" \
-		-extfile "$EASYRSA_TEMP_FILE" -days $EASYRSA_CERT_EXPIRE -batch $opts \
-		|| die "signing failed (openssl output above may have more detail)"
+	if [ -n "${EASYRSA_CERT_END_DATE}" ]
+	then
+		# use explicit end date for "Not After"
+		"$EASYRSA_OPENSSL" ca -utf8 -in "$req_in" -out "$crt_out_tmp" -config "$EASYRSA_SSL_CONF" \
+			-extfile "$EASYRSA_TEMP_FILE" -enddate "$EASYRSA_CERT_END_DATE" -batch $opts \
+			|| die "signing failed (openssl output above may have more detail)"
+	else
+		"$EASYRSA_OPENSSL" ca -utf8 -in "$req_in" -out "$crt_out_tmp" -config "$EASYRSA_SSL_CONF" \
+			-extfile "$EASYRSA_TEMP_FILE" -days $EASYRSA_CERT_EXPIRE -batch $opts \
+			|| die "signing failed (openssl output above may have more detail)"
+	fi
 	mv "$crt_out_tmp" "$crt_out"; EASYRSA_TEMP_FILE_2=
 	notice "\
 Certificate created at: $crt_out
@@ -1123,6 +1132,9 @@ while :; do
 		export EASYRSA_CA_EXPIRE="$val"
 		export EASYRSA_CRL_DAYS="$val"
 		;;
+	--enddate)
+		export EASYRSA_CERT_END_DATE="$val"
+		;;
 	--pki-dir)
 		export EASYRSA_PKI="$val" ;;
 	--use-algo)
